### PR TITLE
SUBMARINE-638. [Spark-Security]Apply global/local limit after masking projection

### PR DESCRIPTION
**Pull request by harsha-tenneti**
### What is this PR for?
This PR optimises the limit query when masking policies are applied in the spark-security plugin. This is applying the global limit on top masking projection if there is a limit in the initial plan

### What type of PR is it?
[Improvement]

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-638

### How should this be tested?
https://travis-ci.org/github/harsha-tenneti/submarine/builds/754430620

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by harsha-tenneti** [comment]
@yaooqinn can you please help look at this PR

File: submarine-security/spark-security/src/main/scala/org/apache/spark/sql/catalyst/optimizer/SubmarineDataMaskingExtension.scala [hunk 1]
```diff
@@ -233,7 +233,20 @@ case class SubmarineDataMaskingExtension(spark: SparkSession) extends Rule[Logic
         case p if hasCatalogTable(p) => SubmarineDataMasking(p)
       }
 
-      marked transformAllExpressions {
+      // Extract global/local limit if any and apply after masking projection
+      val limitExpr: Option[Expression] = plan match {
+        case unaryNode: UnaryNode =>
+          unaryNode match {
```
Comment (yaooqinn) [inline submarine-security/spark-security/src/main/scala/org/apache/spark/sql/catalyst/optimizer/SubmarineDataMaskingExtension.scala:239]: redundant here？

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
also cc @xunliu @jiwq

**Comment by yaooqinn** [comment]
LGTM

**Comment by yaooqinn** [comment]
thanks~ merged to master

**Pull request closed without merging**